core/src/main/java/com/aem/demo/core/models/MainPageModel.java

```diff
@@ -56,7 +55,7 @@ private void init() {
             return;
         }
         models = new ArrayList<>();
-        holdModels = new ArrayList<>();
+        List<NewsCardModel> holdModels = new ArrayList<>();
```

Comment:
See @Zubayr99, your code is working here! But in a usual project, they would rather expect of you the following logic:
1) extract the iteration of all the children of a page into a separate method. Like `getAllModels()`
2) have only one list of models. Namely, `models`. Not two different variables for a list of models.
3) do it like
```
models = searchText != null ? searchService.search(searchText) : getAllModels();
```
4) do pagination. Pagination happens in all cases -- whether there is search text or there isn't, thether there is page number or there isn't.
5) if there's a page number -- then, extract items for the page.
```
models = separateModels(models, pageNum);
```
6) And above all, in a pro project, all the stuff concerning creating the list of models would be also extracted into yet another method.